frontend/src/Editor/LeftSidebar/SidebarGlobalSettings.jsx

```diff
@@ -23,6 +29,19 @@ export const LeftSidebarGlobalSettings = ({
     bottom: '0px',
     left: '0px',
   };
+
+  React.useEffect(() => {
+    setRealState(currentState);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentState.components]);
+
+  React.useEffect(() => {
+    console.log('check', realState);
+    backgroundFxQuery &&
+      globalSettingsChanged('canvasBackgroundColor', resolveReferences(backgroundFxQuery, realState));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [realState]);
```

Comment:
How about this to make the background color change as soon as the value computed by backgroundFxQuery changes?
```suggestion
  }, [JSON.stringify(resolveReferences(backgroundFxQuery, realState))]);
```